ping.go

```diff
@@ -370,15 +372,29 @@ func (p *Pinger) Run() error {
 			}
 		}
 		if p.Count > 0 && p.PacketsRecv >= p.Count {
-			close(p.done)
+			_ = p.Stop()
 			wg.Wait()
 			return nil
 		}
 	}
 }
 
-func (p *Pinger) Stop() {
+func (p *Pinger) Stop() error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	ok := true
+	select {
+	case _, ok = <-p.done:
+	default:
+	}
+
+	if !ok {
+		return ErrTornDown
```

Comment:
What would be the use-case of this error? seems like most people would want to ignore this situation, which is a bit easier and cleaner if we just return `nil`

Reply:
The `error` returned is just an indicator really that I thought _might_ be useful to someone somewhere. It could equally be a true/false boolean flag. Or nothing, as you suggest - I'm not dead set on it by any means. I could perhaps see it as being useful in situations where people are using a pinger in goroutine and might like to use it to help detect race conditions.